File: Services/NavigationService.swift
//
//  NavigationService.swift
//  UDrive
//
//  Created on 2024
//

import SwiftUI

enum AppScreen {
    case onboarding
    case login
    case signupChoice
    case studentDashboard
    case driverDashboard
    case bookRide
    case trackShuttle
    case driverRegistration
    case emergencyRide
}

class NavigationService: ObservableObject {
    @Published var currentScreen: AppScreen = .onboarding
    @Published var isLoggedIn: Bool = false
    @Published var currentUser: User?
    
    func navigate(to screen: AppScreen) {
        currentScreen = screen
    }
    
    func login(user: User) {
        currentUser = user
        isLoggedIn = true
        currentScreen = user.userType == .student ? .studentDashboard : .driverDashboard
    }
    
    func logout() {
        currentUser = nil
        isLoggedIn = false
        currentScreen = .onboarding
    }
}

File: lib/services/navigation_service.dart
import 'package:flutter/foundation.dart';
import '../models/user.dart';

enum AppScreen {
  onboarding,
  login,
  signupChoice,
  studentDashboard,
  driverDashboard,
  bookRide,
  trackShuttle,
  driverRegistration,
  emergencyRide,
}

class NavigationService extends ChangeNotifier {
  AppScreen _currentScreen = AppScreen.onboarding;
  bool _isLoggedIn = false;
  User? _currentUser;

  AppScreen get currentScreen => _currentScreen;
  bool get isLoggedIn => _isLoggedIn;
  User? get currentUser => _currentUser;

  void navigate(AppScreen screen) {
    _currentScreen = screen;
    notifyListeners();
  }

  void login(User user) {
    _currentUser = user;
    _isLoggedIn = true;
    _currentScreen = user.userType == UserType.student
        ? AppScreen.studentDashboard
        : AppScreen.driverDashboard;
    notifyListeners();
  }

  void logout() {
    _currentUser = null;
    _isLoggedIn = false;
    _currentScreen = AppScreen.onboarding;
    notifyListeners();
  }
}

